pkg/crc/preflight/preflight.go

```diff
@@ -0,0 +1,115 @@
+package preflight
+
+import (
+	"fmt"
+	"github.com/code-ready/crc/pkg/crc/output"
+	"github.com/spf13/viper"
+	"os"
+)
+
+type PreflightCheckFixFuncType func() (bool, error)
+
+func preflightCheckSucceedsOrFails(configNameOverrideIfSkipped string, check PreflightCheckFixFuncType, message string, configNameOverrideIfWarning string) {
+	output.Out("-- %s ... ", message)
+	isConfiguredToSkip := viper.GetBool(configNameOverrideIfSkipped)
+	isConfiguredToWarn := viper.GetBool(configNameOverrideIfWarning)
+
+	if isConfiguredToSkip {
+		output.Out("SKIP")
+		return
+	}
+
+	ok, err := check()
+	if ok {
+		output.Out("OK")
+		return
+	}
+
+	output.Out("FAIL")
+	errorMessage := fmt.Sprintf("   %s", err.Error())
+	output.Out(errorMessage)
+	if isConfiguredToWarn {
+		output.Out(errorMessage)
+	} else {
+		os.Exit(1)
+	}
+}
+
+func preflightCheckAndFix(check, fix PreflightCheckFixFuncType, message string) {
+	output.Out("-- %s ... ", message)
+
+	if ok, _ := check(); ok {
+		output.Out("OK")
+		return
+	}
+	ok, err := fix()
+	if ok {
+		output.Out("OK")
+		return
+	}
+
+	output.Out("FAIL")
+	output.Out(err.Error())
+}
+
+// StartPreflightChecks performs the preflight checks before starting the cluster
+func StartPreflightChecks() {
+	preflightCheckSucceedsOrFails("",
+		checkVirtualizationEnabled,
+		"Checking if Virtualization is enabled",
+		"",
+	)
+	preflightCheckSucceedsOrFails("",
+		checkKvmEnabled,
+		"Checking if KVM is enabled",
+		"",
+	)
+	preflightCheckSucceedsOrFails("",
+		checkLibvirtInstalled,
+		"Checking if Libvirt is installed",
+		"",
+	)
+	preflightCheckSucceedsOrFails("",
+		checkUserPartOfLibvirtGroup,
+		"Checking if user is part of libvirt group",
+		"",
+	)
+	preflightCheckSucceedsOrFails("",
+		checkLibvirtEnabled,
+		"Checking if Libvirt is enabled",
+		"",
+	)
+	preflightCheckSucceedsOrFails("",
+		checkLibvirtServiceRunning,
+		"Checking if Libvirt daemon is running",
+		"",
+	)
+	preflightCheckSucceedsOrFails("",
+		checkDockerMachineDriverKvmInstalled,
+		"Checking if docker-machine-driver-kvm is installed",
+		"",
+	)
+	preflightCheckSucceedsOrFails("",
+		checkDefaultPoolAvailable,
+		"Checking if default pool is available",
+		"",
+	)
+	preflightCheckSucceedsOrFails("",
+		checkDefaultPoolHasSufficientSpace,
+		"Checking if default pool has sufficient free space",
+		"",
+	)
+}
+
+// SetupHost performs the prerequisite checks and setups the host to run the cluster
+func SetupHost() {
+	preflightCheckAndFix(checkVirtualizationEnabled, fixVirtualizationEnabled, "Setting up virtualization")
```

Comment:
format both blocks in the same way... else it looks VERY different and unclear what happens (difference is).